### PR TITLE
Added the hook actionProductImageDeleted

### LINKED ISSUE #23987: Execute some hooks when images are added/removed
Images is an important topic when merchant has a lot of products, especially if he uses high size images and large image types. In this case, storing them in local drive is not cost effective and not reliabe.

Using cloud storage services (like Azure or S3 to name of few) is a good solution.They also improve performance by offloading web server from image requests.

Also migrating to another host is more easy, because the files backup is smaller. One approach is to keep only `tmp` folder inside `img` folder (just for backoffice images)  and upload all the rest to the cloud.

The problem is for adding/ removal of images. There should be a syncing mechanism between local drive and cloud.

It is possible to upload product image to the cloud just after the upload to local drive using the hook `actionWatermark`.

But for deletion, I don’t find a hook that executes once a product image is deleted.

For the other images (manufacturers, suppliers, shipping... ) I dont find any hooks executed on add/removal of images.

I understand that the most important images in size are product images (and maybe manufacturers). But when we activate media servers in `Advanced >performance`, all images in front office are from cloud, which emphasizes the need of a syncing mechanism also for categories, suppliers..etc.

**Suggestions:**
Add hook actionProductImageDeleted
Add hook actionManufacturerImageAdded
Add hook actionManufacturerImageDeleted
Add other similar hooks for categories….

I would like to know pros/cons of adding such hooks.
Community feedback is much appreciated.

### PR DESCRIPTION
**Pull request by arouiadib**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | This PR adds a hook named actionProductImageDeleted that runs when a product image is deleted.
| Type?             |  improvement 
| Category?         | CO 
| BC breaks?        | no
| Deprecations?     |  no
| Fixed ticket?     | Fixes #23987
| How to test?      | 
| Possible impacts? | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24051)

### CODE REVIEW

File: controllers/admin/AdminProductsController.php [hunk 1]
```diff
@@ -1557,6 +1557,8 @@ public function ajaxProcessDeleteProductImage($id_image = null)
             $res &= @unlink(_PS_TMP_IMG_DIR_ . 'product_mini_' . $image->id_product . '_' . $this->context->shop->id . '.jpg');
         }
 
+        Hook::exec('actionProductImageDeleted', ['id_image' => $image->id]);
+
```
Comment (PululuK) [inline controllers/admin/AdminProductsController.php]: We are not sure that image has been deleted. Place it just before 
```php
$this->jsonConfirmation($this->_conf[7]);

Comment (arouiadib) [reply controllers/admin/AdminProductsController.php]: great point @PululuK . Thanks for your time

Comment (Progi1984) [reply controllers/admin/AdminProductsController.php]: Approved with @PululuK

**Review by Progi1984 — COMMENTED** [review]
Please follow indication from @PululuK

**Review by arouiadib — COMMENTED** [review]
Done. Thanks

**Review by PululuK — DISMISSED** [review]
Thanks for your contribution @arouiadib !
If you have the time, please add doc. 
You can see this example : https://github.com/PrestaShop/docs/pull/770

**Comment by arouiadib** [comment]
Yes @PululuK , forgot this point . Thanks for the example.

**Review by Progi1984 — DISMISSED** [review]

**Comment by Progi1984** [comment]
@arouiadib A last thing : Could you define the `How to test?` like in #21025 ?

**Comment by arouiadib** [comment]
@Progi1984 , thanks for your guidance. I will do asap

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by arouiadib** [comment]
Docs are added [here](https://github.com/PrestaShop/docs/pull/957)  as of @PululuK CR.

I am still trying to add a `how to test?` section as of @Progi1984 CR, but it is a bit delicate because the image deletion happens in an ajax call and there is no direct way showing something while the hook executes. What if I give a module in github that registers this hook and do something?!

**Comment by PululuK** [comment]
Hi @arouiadib 

Yes you can provide a basic module with this hook implementation.

```php
   public function hookActionProductImageDeleted($params){
         var_dump($params);
         die(); 
  }
```

**Comment by arouiadib** [comment]
> Hi @arouiadib
> 
> Yes you can provide a basic module with this hook implementation.
> 
> ```
>    public function hookActionProductImageDeleted($params){
>          var_dump($params);
>          die(); 
>   }
> ```

Ok great @PululuK . I will do asap.

**Comment by NeOMakinG** [comment]
Isn't the QA module enough to test this ? https://github.com/PrestaShop/ps_qualityassurance

**Review by NeOMakinG — DISMISSED** [review]

**Comment by eternoendless** [comment]
This PR is likely to become obsolete after the Product page is completely migrated. Perhaps it will make more sense to add the hook in the new page?

ping @matks @jolelievre

**Comment by NeOMakinG** [comment]
> This PR is likely to become obsolete after the Product page is completely migrated. Perhaps it will make more sense to add the hook in the new page?
> 
> ping @matks @jolelievre

Right, forgot that this page (on develop) shouldn't be there in the future

**Comment by arouiadib** [comment]
Thanks @eternoendless , nice info. Time to discover the new page :).

Thanks @NeOMakinG 

The most important thing for me was: to learn how to properly add a hook to PS core.

File: install-dev/upgrade/sql/1.7.8.0.sql [hunk 2]
```diff
@@ -22,6 +22,7 @@ INSERT IGNORE INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `po
   (NULL, 'displayAdminGridTableBefore', 'Display before Grid table', 'This hook adds new blocks before Grid component table.', '1'),
   (NULL, 'displayAdminGridTableAfter', 'Display after Grid table', 'This hook adds new blocks after Grid component table.', '1'),
   (NULL, 'displayAdminOrderCreateExtraButtons', 'Add buttons on the create order page dropdown', 'Add buttons on the create order page dropdown', '1'),
+  (NULL, 'actionProductImageDeleted', 'Runs an action when a product image is deleted', 'Runs an action when a product image is deleted', '1'),
```
Comment (Progi1984) [inline install-dev/upgrade/sql/1.7.8.0.sql]: Could you move your query in `1.7.9.0.sql` ?

**Review by Progi1984 — COMMENTED** [review]
A last feedback

**Comment by arouiadib** [comment]
@Progi1984 feedbacks are the best thing in this repo.

I have applied your CR, although I didn't really understood why? :sweat_smile: 

Any explanations about branches and releases are most welcome.

Thanks a lot for your time

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
@arouiadib A last thing : Could you define the `How to test?` like in #21025 ?

**Comment by PierreRambaud** [comment]
Hi @arouiadib ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**